// @cloudinary/url-gen/actions/effect/Cartoonify.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var tslib_es6 = require('../../tslib.es6-18ed4f64.cjs');
var internal_qualifier_Qualifier = require('../../internal/qualifier/Qualifier.cjs');
var internal_qualifier_QualifierValue = require('../../internal/qualifier/QualifierValue.cjs');
var internal_Action = require('../../internal/Action.cjs');
require('../../qualifiers/flag/FlagQualifier.cjs');
require('../../internal/utils/dataStructureUtils.cjs');

/**
 * @description Applies a cartoon effect to an image.
 * @extends SDK.Action
 * @memberOf Actions.Effect
 * @see Visit {@link Actions.Effect|Effect} for an example
 */
var CartoonifyEffect = /** @class */ (function (_super) {
    tslib_es6.__extends(CartoonifyEffect, _super);
    function CartoonifyEffect(effectName, strength) {
        var _this = 
        // We don't pass level in the constructor, we'll build it in the prepareParam
        _super.call(this) || this;
        _this.cartoonifyStrength = strength;
        _this.effectName = effectName;
        return _this;
    }
    /**
     * @description Sets the thickness of the lines.
     * @param {number} lineStrength The thickness of the lines. (Range: 0 to 100, Server default: 50)
     * @return {this}
     */
    CartoonifyEffect.prototype.lineStrength = function (lineStrength) {
        this.cartoonifyStrength = lineStrength;
        return this;
    };
    /**
     * @description Achieves a black and white cartoon effect.
     * @return {this}
     */
    CartoonifyEffect.prototype.blackwhite = function () {
        this.colorReduction = 'bw';
        return this;
    };
    /**
     * @description
     * Sets the decrease in the number of colors and corresponding saturation boost of the remaining colors. <br/>
     * Higher reduction values result in a less realistic look.
     * @param {number } level The decrease in the number of colors and corresponding saturation boost of the remaining colors. (Range: 0 to 100, Server default: automatically adjusts according to the line_strength value). Set to 'bw' for a black and white cartoon effect.
     * @return {this}
     */
    CartoonifyEffect.prototype.colorReductionLevel = function (level) {
        this.colorReduction = level;
        return this;
    };
    CartoonifyEffect.prototype.prepareQualifiers = function () {
        this.addQualifier(new internal_qualifier_Qualifier.Qualifier('e', new internal_qualifier_QualifierValue.QualifierValue([this.effectName, this.cartoonifyStrength, this.colorReduction])));
        return;
    };
    return CartoonifyEffect;
}(internal_Action.Action));

exports.CartoonifyEffect = CartoonifyEffect;
